src/diamond/handler/elasticsearchHandler.py

```diff
@@ -0,0 +1,232 @@
+# coding=utf-8
+
+"""
+Send metrics to an
+[ElasticSearch](https://www.elastic.co/products/elasticsearch) cluster.
+
+ElasticSearch is an easily scalable database with a REST/JSON API. It has
+been used for log storage and analysis and now for metrics as well!
+
+ElasticSearch has introduced the
+[Beats](https://www.elastic.co/products/beats) library to ship various metrics
+to ES. This handler is Beats compatible.
+
+#### Dependencies
+
+  * elasticsearch python module (pip install elasticsearch)
+
+#### Setup
+
+1. Decide what to call the index the metrics will be saved in. By default
+the handler follows the statsd and beats tradition and creates daily indices
+called "diamond-YYYY.MM.DD". Kibana, the most common graphical interface
+to ElasticSearch has built-in support for this.
+
+2. Decide if you want to change the default field names.
+
+2. Load a default template into ES to define the fields and options:
+
+```
+curl -XPUT localhost:9200/_template/diamond-template -d '
+{
+  "template": "diamond-*",
+  "mappings": {
+    "_default_": {
+      "_all": {
+        "enabled": false
+      },
+      "dynamic_templates": [
+        {
+          "template1": {
+            "mapping": {
+              "doc_values": true,
+              "ignore_above": 1024,
+              "index": "not_analyzed",
+              "type": "{dynamic_type}"
+            },
+            "match": "*"
+          }
+        }
+      ],
+      "properties": {
+        "timestamp": {
+          "type": "date"
+        }
+      }
+    }
+  },
+  "settings": {
+    "index.refresh_interval": "5s"
+  }
+}'
+```
+
+4. Add the ElasticSearchHandler config to the diamond.conf handlers section:
+(minimum config shown)
+
+```
+handlers = diamond.handler.elasticsearchHandler.ElasticSearchHandler
+
+[handlers]
+
+[[ElasticSearchHandler]]
+host_urls = http://127.0.0.1:9200/
+```
+
+#### Acknowledgements
+
+ElasticSearchHandler was made possible by [ACT.md](http://act.md)
+
+"""
+
+__author__ = 'imre Fitos'
+__version__ = '1.0'
+
+import collections
+import time
+from Handler import Handler
+
+try:
+    from elasticsearch import Elasticsearch, helpers
+except ImportError:
+    Elasticsearch = None
+
+
+class ElasticSearchHandler(Handler):
+
+    def __init__(self, *args, **kwargs):
+        super(ElasticSearchHandler, self).__init__(*args, **kwargs)
+
+        if Elasticsearch is None:
+            self.log.error(
+                'Failed to import elasticsearch module. Handler disabled')
+            self.enabled = False
+            return
+        # attempt to connect
+        try:
+            url_list = []
+            for host in self.config['host_urls'].split(','):
+                url_list.append(host)
+            self.es = Elasticsearch(url_list)
+        except Exception as exc:
+            self.log.error("ElasticSearchHandler start failed: {}".format(exc))
+            self.enabled = False
+        # save options into vars
```

Comment:
Please spread out your code a little to make it easier to read. A newline above here might help.